src/runtime/virtcontainers/clh.go

```diff
@@ -520,9 +520,6 @@ func (clh *cloudHypervisor) CreateVM(ctx context.Context, id string, network Net
 
 	// Make sure the kernel path is valid if no igvm set
 	if igvmPath == "" {
-		if clh.config.ConfidentialGuest {
```

Comment:
Suggestion: Check for sev_snp_guest presence instead of confidential_ugest

Reply:
Isn't the goal to be able to run the confidential_guest path without using sev-snp?

Reply:
Yes, that is the goal, but as far as I can see this check mandates IGVM when using confidential guest. It is clear that we want to remove this. However, we should still mandate IGVM when sev_snp_guest is set to true, because this path will clearly not work w/o IGVM